### ISSUE TITLE
forward-mode AD coverage request for PyTorch 1.12

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#434]
The following operations fail the functorch jvp tests. This is probably because they do not have coverage in pytorch/pytorch, but could also be due to a functorch problem or a bug in pytorch/pytorch forward-mode AD testing. This issue is asking for:
1. implement the forward-mode AD formula in pytorch/pytorch
2. fix the forward-mode AD formula in pytorch/pytorch
- [x] 'torch.atan2',
- [x] 'torch.clamp',
- [x] 'torch.dist', https://github.com/pytorch/pytorch/pull/74205
- [x] 'torch.logsumexp',
- [x] 'torch.max', bug in formula in PyTorch: https://github.com/pytorch/pytorch/issues/69034
- [x] 'torch.min', bug in formula in PyTorch: https://github.com/pytorch/pytorch/issues/69034
- [x] 'torch.nn.functional.cross_entropy',
- [ ] 'torch.nn.functional.embedding',
- [x] 'torch.nn.functional.log_softmax',
- [x] 'torch.nn.functional.nll_loss',
- [ ] 'torch.nn.functional.prelu',
- [x] 'torch.nn.functional.softmax',
- [x] 'torch.norm' https://github.com/pytorch/pytorch/pull/74205
- [x] 'torch.nn.functional.batch_norm',  (NB: not in https://github.com/pytorch/pytorch/issues/71117)
  - [x] needs functorch testing fix (batch_norm saved mean / saved var get dual tensors)
  - [x] needs better error message in pytorch/pytorch: https://github.com/pytorch/pytorch/issues/73541
- [x] 'nn.functional.conv_transpose3d': TODO(rzou): precision issue
- [x] 'torch.nn.functional.max_pool1d', (NB: not in https://github.com/pytorch/pytorch/issues/71117)
  - this is a composite compliance issue TODO(rzou)
- [x] 'torch.div', (NB: not in https://github.com/pytorch/pytorch/issues/71117)
  - fixed in pytorch/pytorch

**Comment by soulitzer** [comment]
Update: https://github.com/pytorch/pytorch/pull/73741 (logsumexp, cross_entropy, log_softmax, nll_loss, softmax) has landed and https://github.com/pytorch/pytorch/pull/74042 (clamp), https://github.com/pytorch/pytorch/pull/74205 (norm, dist) are on the way. That leaves prelu, embedding and atan2 remaining in terms of new formulas that need to be added.
